Source/EasyNetQ/Persistent/PersistentChannel.cs

```diff
@@ -46,13 +47,71 @@ IEventBus eventBus
     }
 
     /// <inheritdoc />
-    public async Task<TResult> InvokeChannelActionAsync<TResult, TChannelAction>(
+    public ValueTask<TResult> InvokeChannelActionAsync<TResult, TChannelAction>(
         TChannelAction channelAction, CancellationToken cancellationToken = default
     ) where TChannelAction : struct, IPersistentChannelAction<TResult>
     {
         if (disposed)
             throw new ObjectDisposedException(nameof(PersistentChannel));
 
+        cancellationToken.ThrowIfCancellationRequested();
+
+        return TryInvokeChannelActionFast<TResult, TChannelAction>(channelAction, out var result)
```

Comment:
I suppose double Acquire call to lock should not that slow in high concurrency cases, because internally Semaphore.Wait/Semaphore.WaitAsync methods check volatile counter under the hood.